app/backend/src/couchers/interceptors.py

```diff
@@ -89,56 +94,64 @@ def intercept_service(self, continuation, handler_call_details):
         return continuation(handler_call_details)
 
 
-class LoggingInterceptor(grpc.ServerInterceptor):
+class MonitoringInterceptor(grpc.ServerInterceptor):
     """
     Measures and logs the time it takes to service each incoming call.
-    """
-
-    def intercept_service(self, continuation, handler_call_details):
-        handler = continuation(handler_call_details)
-        prev_func = handler.unary_unary
-        method = handler_call_details.method
-
-        def timetaking_function(request, context):
-            if LOG_VERBOSE_PB:
-                logger.info(f"Got request: {method}. Request: {request}")
-            else:
-                logger.info(f"Got request: {method}")
-            start = perf_counter_ns()
-            res = prev_func(request, context)
-            finished = perf_counter_ns()
-            duration = (finished - start) / 1e6  # ms
-            if LOG_VERBOSE_PB:
-                logger.info(f"Finished request (in {duration:0.2f} ms): {method}. Response: {res}")
-            else:
-                logger.info(f"Finished request (in {duration:0.2f} ms): {method}")
-            return res
 
-        return grpc.unary_unary_rpc_method_handler(
-            timetaking_function,
-            request_deserializer=handler.request_deserializer,
-            response_serializer=handler.response_serializer,
-        )
-
-
-class ErrorSanitizationInterceptor(grpc.ServerInterceptor):
+    If the call resulted in a non-gRPC error, this also strips away the error details and returns a generic error.
     """
-    If the call resulted in a non-gRPC error, this strips away the error details.
 
-    It's important to put this first, so that it does not interfere with other interceptors.
-    """
+    def _sanitized_bytes(self, proto):
+        """
+        Remove fields marked sensitive and return serialized bytes
+        """
+        if not proto:
+            return None
+        new_proto = deepcopy(proto)
+        for name, descriptor in new_proto.DESCRIPTOR.fields_by_name.items():
+            if descriptor.GetOptions().Extensions[annotations_pb2.sensitive]:
+                new_proto.ClearField(name)
+        return new_proto.SerializeToString()
+
+    def _store_log(self, method, status_code, duration, user_id, request, response, traceback):
+        req_bytes = self._sanitized_bytes(request)
+        res_bytes = self._sanitized_bytes(response)
+        with session_scope() as session:
+            session.add(
+                APICall(
+                    method=method,
+                    status_code=status_code,
+                    duration=duration,
+                    user_id=user_id,
+                    request=req_bytes,
+                    response=res_bytes,
+                    traceback=traceback,
+                )
+            )
+        logger.debug(f"{user_id=}, {method=}, {duration=} ms, req={len(req_bytes or b'')}, res={len(res_bytes or b'')}")
```

Comment:
The length of the sanitized request and response doesn't add any statistics or debugging aid.

Reply:
I'll remove them